asciinema/recorder.py

```diff
@@ -50,43 +59,48 @@ def record(
     if title:
         full_metadata["title"] = title
 
-    time_offset = 0
+    time_offset: float = 0
 
-    if append and os.stat(path).st_size > 0:
-        time_offset = v2.get_duration(path)
+    if append and os.stat(path_).st_size > 0:
+        assert time_offset is not None
```

Comment:
Sorry I phrased things poorly there. `assert` statements should _not_ be relied upon to impact program behavior or as control flow or as a security check because they _may_ be removed depending on the compiler options as you point out. I doubt that these assertions are costly, but honestly I'm not an expert on the internals of CPython. I think the best approach may be to leave them for now as indicators to be refactored out later as certain behaviors are changed to become more deterministic or handle a specific error case.

Reply:
Got it. Sounds good 👍